### PR TITLE
[pkg] add a debug pickler

### PR DESCRIPTION
**Pull request by PaliC**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70967
* #70966
* #70965
* __->__ #70646
* #70642
* #70641

Adds a stack trace to the pickler in order to help locate deeply nested unpicklable objects. This pr specifically sets up testing and the overarching stack trace (as well as the objects that will be used in it). The debug pickler is also able to find unpicklable objects in dicts.

Differential Revision: [D33414363](https://our.internmc.facebook.com/intern/diff/D33414363)

### CODE REVIEW

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

File: torch/package/_package_pickler.py [hunk 1]
```diff
@@ -93,6 +117,178 @@ def save_global(self, obj, name=None):
     dispatch[FunctionType] = save_global
 
 
+class DebugInfo:
```
Comment (suo) [inline torch/package/_package_pickler.py:120]: Let's split this out into a separate file

**Review by suo — APPROVED** [review]
nice!

**Review by suo — APPROVED** [review]

**Review by suo — APPROVED** [review]
nice :)

**Comment by PaliC** [comment]
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33414363).

**Pull request closed without merging**